staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -151,7 +151,12 @@ func ValueOrDefault(intOrPercent *IntOrString, defaultValue IntOrString) *IntOrS
 	return intOrPercent
 }
 
-func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+//GetScaledValueFromIntOrPercent is meant to replace GetValueFromIntOrPercent.
+//This method returns a scaled value from an IntOrString type. If the IntOrString
+//is a percentage string value it's treated as a percentage and scaled appropriately
+//in accordance to the total, if it's an int value it's treated as a a simple value and
+//if it is a numeric string value it returns an error.
+func GetScaledValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
```

Comment:
Can we delete the roundUp parameter and return a float?

Since we have to change call sites anyway, and the "rounding" done here is not good :)

Reply:
Wait, I think I misunderstood what this does given that "total" is passed in. I have to run and will look more later.

Reply:
OK I see now that I completely misunderstood what this function does, so sorry for the noise. Given that this is returning a specific number of things, the "scaled" name and rounding makes more sense.

The rounding still doesn't make a lot of sense abstractly, because actually choosing between floor and ceil is not the same as rounding. But I won't ask for a change, since it makes some sense if you're consuming discreet thingies.